### ISSUE TITLE
Viewer crash for long pdf with specific window width

### ISSUE THREAD

**Issue by xiaody** [react-pdf-viewer/react-pdf-viewer#1179]
I created a reproduction repo https://github.com/xiaody/react-pdf-viewer-issue . It's a CRA project with App.js edited to 
```js
import * as React from 'react';

// Core viewer
import { Viewer, Worker } from '@react-pdf-viewer/core';

// Import styles
import '@react-pdf-viewer/core/lib/styles/index.css';

<Viewer fileUrl="/assets/pdf-open-parameters.pdf" />;

export default function App() {
  return (
    <Worker workerUrl='https://ccsp.gtimg.com/ccsp/js/v2.0/worker/pdf.worker-es5.min.js'>
      <Viewer fileUrl="/pytest.pdf" />
    </Worker>
  );
}
```

The file `pytest.pdf` has 200+ pages, weight 933KB.

Clone the repo and run `npm start`. When your window is at specific width, chances are that you see `Uncaught Error: Maximum update depth exceeded.`

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/1246451/173723142-dee40a5a-1c2c-4919-a988-60862d28096e.png">

With a debugger in the `measureRef` function, I can see the `setCacheMeasure` function is called many times. Not sure if it's the cause.

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/1246451/173724083-c26eff32-7252-4c18-b21a-ec3d9f077fd2.png">

**Comment by Garwin4j** [comment]
Having the same issue.

**Comment by Garwin4j** [comment]
@xiaody What seems to make it more stable for me is to set the `defaultScale` : 
```
<div style={{ width: "95vw" }}>
<Viewer fileUrl={props.document.url}
        defaultScale={1}
....
```

**Comment by phuocng** [comment]
This should be fixed in the latest code. I will include the fix in the next version.

**Comment by phuocng** [comment]
Fixed in [v3.9.0](https://react-pdf-viewer.dev/updates/)
